app/cni/cmd/kuma-cni/main.go

```diff
@@ -0,0 +1,266 @@
+}
+
+type PluginConf struct {
+	types.NetConf // You may wish to not nest this type
+
+	RawPrevResult *map[string]interface{} `json:"prevResult"`
+	PrevResult    *current.Result         `json:"-"`
+
+	// plugin-specific fields
+	LogLevel   string     `json:"log_level"`
+	Kubernetes Kubernetes `json:"kubernetes"`
+}
+
+// K8sArgs is the valid CNI_ARGS used for Kubernetes
+// The field names need to match exact keys in kubelet args for unmarshalling
+type K8sArgs struct {
+	types.CommonArgs
+	IP                         net.IP
+	K8S_POD_NAME               types.UnmarshallableString // nolint: golint, stylecheck
+	K8S_POD_NAMESPACE          types.UnmarshallableString // nolint: golint, stylecheck
+	K8S_POD_INFRA_CONTAINER_ID types.UnmarshallableString // nolint: golint, stylecheck
+}
+
+// parseConfig parses the supplied configuration (and prevResult) from stdin.
+func parseConfig(stdin []byte) (*PluginConf, error) {
+	conf := PluginConf{}
+
+	if err := json.Unmarshal(stdin, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse network configuration: %v", err)
+	}
+
+	// Parse previous result. Remove this if your plugin is not chained.
+	if conf.RawPrevResult != nil {
+		resultBytes, err := json.Marshal(conf.RawPrevResult)
+		if err != nil {
+			return nil, fmt.Errorf("could not serialize prevResult: %v", err)
+		}
+		res, err := version.NewResult(conf.CNIVersion, resultBytes)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse prevResult: %v", err)
+		}
+		conf.RawPrevResult = nil
+		conf.PrevResult, err = current.NewResultFromResult(res)
+		if err != nil {
+			return nil, fmt.Errorf("could not convert result to current version: %v", err)
+		}
+	}
+	// End previous result parsing
+
+	return &conf, nil
+}
+
+// cmdAdd is called for ADD requests
+func cmdAdd(args *skel.CmdArgs) error {
+	ctx := context.Background()
+	conf, err := parseConfig(args.StdinData)
+	if err != nil {
+		log.Error(err, "error parsing kuma-cni cmdAdd config")
+		return err
+	}
+	logPrevResult(conf)
+
+	// Determine if running under k8s by checking the CNI args
+	k8sArgs := K8sArgs{}
+	if err := types.LoadArgs(args.Args, &k8sArgs); err != nil {
+		log.Error(err, "error loading kuma-cni cmdAdd args")
+		return err
+	}
+	logContainerInfo(args.Args, args.ContainerID, k8sArgs)
+
+	if string(k8sArgs.K8S_POD_NAMESPACE) == "" || string(k8sArgs.K8S_POD_NAME) != "" {
+		log.Info("no kubernetes data")
+		return prepareResult(conf)
+	}
+
+	if shouldExcludePod(conf.Kubernetes.ExcludeNamespaces, k8sArgs.K8S_POD_NAMESPACE) {
+		log.Info("pod excluded")
+		return prepareResult(conf)
+	}
+
+	containers, initContainersMap, annotations, err := getPodInfoWithRetries(ctx, conf, k8sArgs)
+	if err != nil {
+		log.Info("error getting pod info")
+		return err
+	}
+
+	if isInitContainerPresent(initContainersMap, k8sArgs) {
+		log.Info("pod excluded - init container present")
+		return prepareResult(conf)
+	}
+
+	if containers < 2 {
+		log.Info("not enough containers in pod")
```

Comment:
Can we be a bit more specific? It took me a minute to figure out that we require `kuma-sidecar` to be present.